### PR TITLE
Add support for the Elastic license

### LINKED ISSUE #1: Support Elastic License as well as Apache 2.0
## Description

Currently `go-licenser` only enforces / lints the hardcoded `main.Header` that is in fact an Apache 2.0 license. The upcoming Elastic License should also be supported

### PR DESCRIPTION
**Pull request by exekias**
## Description
 
Introduces support for multiple licenses through the `--license` flag, keeping the
current one (`ASL2`) as the default, and adding an `Elastic` option.

Closes #1

### CODE REVIEW

**Comment by exekias** [comment]
something is wrong after rebasing, will have a look

**Comment by exekias** [comment]
This should be ready now, sorry for the noise :innocent:

File: main.go [hunk 1]
```diff
@@ -102,25 +111,33 @@ func init() {
...
 	}
 
 	os.Exit(Code(err))
 }
 
-func run(args, exclude []string, ext string, dry bool, out io.Writer) error {
+func run(args []string, license string, exclude []string, ext string, dry bool, out io.Writer) error {
+	header, ok := Headers[license]
+	if !ok {
+		return &Error{err: fmt.Errorf("unkown license: %s", license), code: errUnknownLicense}
```
Comment (ph) [inline main.go]: s/unkown/unknown/

Comment (ph) [reply main.go]: nit: Maybe we could add the valid licenses? This will make the error message easily actionable.

**Comment by exekias** [comment]
Thank you for the reviews @marclop  @ph! I pushed a fix to the flag docs, also updated README

**Pull request merged**